qiskit/transpiler/__init__.py

```diff
@@ -36,84 +36,164 @@
-#. ``translation`` - This stage translates the gates in the circuit to the target backend's basis set.
-   See :ref:`translation_stage` for more details.
-#. ``optimization`` - This stage runs the main optimization loop repeatedly
-   until a condition (such as fixed depth) is reached. See :ref:`optimization_stage` for more details.
-#. ``scheduling`` - This stage is for any hardware-aware scheduling passes. See
-   :ref:`scheduling_stage` for more details.
-
-When using :func:`~.transpile`, the implementation of each stage can be modified with the ``*_method``
-arguments (e.g. ``layout_method``). These can be set to one of the built-in methods and
-can also refer to available external plugins. See
-:mod:`qiskit.transpiler.preset_passmanagers.plugin` for details on this plugin interface.
-
-.. _working_with_preset_pass_managers:
-
-Working with Preset Pass Managers
-=================================
-
-Qiskit includes functions to build preset :class:`~.PassManager` objects.
-These preset passmanagers are used by the :func:`~.transpile` function
-for each optimization level. There are 4 optimization levels ranging from 0 to 3, where higher
-optimization levels take more time and computational effort but may yield a
-more optimal circuit.
-Optimization level 0 is intended for device characterization experiments and, as such, only
-maps the input circuit to the constraints of the target backend, without
-performing any optimizations. Optimization level 3 spends the most effort to optimize the circuit.
-However, as many of the optimization techniques in the transpiler are heuristic based, spending more
-computational effort does not always result in an improvement in the quality of the output
-circuit.
+Qiskit uses the graph-based :class:`.DAGCircuit` intermediate representation (IR) of a circuit
+throughout the transpiler stack, rather than the tree-based :class:`.QuantumCircuit`.  A transpiler
+pipeline is a :class:`.PassManager` object, whose :meth:`.PassManager.run` method takes in a
+:class:`.QuantumCircuit` and converts it to a :class:`.DAGCircuit`, then subjects the IR to a
+sequence of *passes*, finally returning a :class:`.QuantumCircuit` back.  A pass is either an
+:class:`.AnalysisPass`, which calculates and stores properties about the circuit in the
+stateful :class:`.PropertySet`, or a :class:`.TransformationPass`, which modifies the IR
+to achieve a particular singular goal.  You can think of a pipeline as being split into
+"stages", where each stage is responsible for one high-level transformation.
+
+Qiskit exposes a default transpilation pipeline builder using the function
+:func:`.generate_preset_pass_manager`.  This returns a properly configured pipeline for complete
+transpilation, at a chosen ``optimization_level`` (between 0 and 3, inclusive).  Unless you are
+looking for something highly specialized, this is almost certainly the entry point you want.  A
+sample transpilation looks like:
```

Comment:
(snippet not formatted)

```suggestion
sample transpilation looks like::
```